src/Context.php

```diff
@@ -32,6 +32,14 @@ class Context extends AbstractKeyValueStore
         'hex',
     ];
 
+    /**
+     * Do not propogate value of $match.
+     */
+    public function __clone()
+    {
+        unset($this->data['$match']);
```

Comment:
Had a couple ideas for removing Conditional logic from Context; this is my favorite at the moment.

1) Add optional $removeOnClone parameter (or $isTemporary) to `\Magento\Upward\AbstractKeyValueStore::set`, default to false
2) Keep track of those keys and move this logic to `AbstractKeyValueStore` (loop through keys and unset on clone)

Reply:
I think I like this better than what I'd done too; it gets us away from coding "magic keys" into the `Context` class